net/dns/resolver/forwarder.go

```diff
@@ -698,7 +696,18 @@ func (f *forwarder) forwardWithDestChan(ctx context.Context, query packet, respo
 		resolvers = f.resolvers(domain)
 		if len(resolvers) == 0 {
 			metricDNSFwdErrorNoUpstream.Add(1)
-			return errNoUpstreams
+			f.logf("no upstream resolvers set, returning SERVFAIL")
+			res, err := servfailResponse(query)
+			if err != nil {
+				f.logf("building servfail response: %v", err)
+				return nil
```

Comment:
why's this nil? add a comment? it looks like a mistake otherwise.

Reply:
Sent https://github.com/tailscale/tailscale/pull/5326